Build issues with 2.4 on Linux

Hello,
two users reported a problem compiling this plugin and I could reproduce it.
While linking the compiled code it errors out because of multiple definitions.
The complete log:
```
which gimptool-2.0 && \
gcc -o ./bin/bimp -Wall -O2 -Wno-unused-variable -Wno-pointer-sign src/*.c src/manipulation-gui/*.c src/images/*.c -I/usr/include/gimp-2.0 -I/usr/include/gdk-pixbuf-2.0 -I/usr/include/glib-2.0 -I/usr/lib/glib-2.0/include -I/usr/include/libmount -I/usr/include/blkid -I/usr/include/cairo -I/usr/include/pixman-1 -I/usr/include/freetype2 -I/usr/include/libpng16 -I/usr/include/harfbuzz -I/usr/include/gegl-0.4 -I/usr/include/gio-unix-2.0 -I/usr/include/json-glib-1.0 -I/usr/include/babl-0.1 -I/usr/include/gtk-2.0 -I/usr/lib/gtk-2.0/include -I/usr/include/pango-1.0 -I/usr/include/fribidi -I/usr/include/atk-1.0 -pthread  -lgimpui-2.0 -lgimpwidgets-2.0 -lgimpmodule-2.0 -lgimp-2.0 -lgimpmath-2.0 -lgimpconfig-2.0 -lgimpcolor-2.0 -lgimpbase-2.0 -lgegl-0.4 -lgegl-npd-0.4 -lm -Wl,--export-dynamic -lgmodule-2.0 -pthread -lglib-2.0 -ljson-glib-1.0 -lbabl-0.1 -lgtk-x11-2.0 -lgdk-x11-2.0 -lpangocairo-1.0 -latk-1.0 -lcairo -lgdk_pixbuf-2.0 -lgio-2.0 -lpangoft2-1.0 -lpango-1.0 -lgobject-2.0 -lglib-2.0 -lharfbuzz -lfontconfig -lfreetype  -lm -DGIMP_DISABLE_DEPRECATED
/usr/bin/gimptool-2.0
In file included from /usr/include/gtk-2.0/gtk/gtkobject.h:37,
                 from /usr/include/gtk-2.0/gtk/gtkwidget.h:36,
                 from /usr/include/gtk-2.0/gtk/gtkcontainer.h:35,
                 from /usr/include/gtk-2.0/gtk/gtkbin.h:35,
                 from /usr/include/gtk-2.0/gtk/gtkwindow.h:36,
                 from /usr/include/gtk-2.0/gtk/gtkdialog.h:35,
                 from /usr/include/gtk-2.0/gtk/gtkaboutdialog.h:32,
                 from /usr/include/gtk-2.0/gtk/gtk.h:33,
                 from src/bimp-gui.c:5:
/usr/include/gtk-2.0/gtk/gtktypeutils.h:236:1: warning: 'GTypeDebugFlags' is deprecated [-Wdeprecated-declarations]
  236 | void            gtk_type_init   (GTypeDebugFlags    debug_flags);
      | ^~~~
In file included from /usr/include/glib-2.0/gobject/gobject.h:24,
                 from /usr/include/glib-2.0/gobject/gbinding.h:29,
                 from /usr/include/glib-2.0/glib-object.h:22,
                 from /usr/include/glib-2.0/gio/gioenums.h:28,
                 from /usr/include/glib-2.0/gio/giotypes.h:28,
                 from /usr/include/glib-2.0/gio/gio.h:26,
                 from /usr/include/gtk-2.0/gdk/gdkapplaunchcontext.h:30,
                 from /usr/include/gtk-2.0/gdk/gdk.h:32,
                 from /usr/include/gtk-2.0/gtk/gtk.h:32,
                 from src/bimp-gui.c:5:
/usr/include/glib-2.0/gobject/gtype.h:679:1: note: declared here
  679 | {
      | ^
In file included from /usr/include/gtk-2.0/gtk/gtktoolitem.h:31,
                 from /usr/include/gtk-2.0/gtk/gtktoolbutton.h:30,
                 from /usr/include/gtk-2.0/gtk/gtkmenutoolbutton.h:30,
                 from /usr/include/gtk-2.0/gtk/gtk.h:126,
                 from src/bimp-gui.c:5:
/usr/include/gtk-2.0/gtk/gtktooltips.h:73:3: warning: 'GTimeVal' is deprecated: Use 'GDateTime' instead [-Wdeprecated-declarations]
   73 |   GTimeVal last_popdown;
      |   ^~~~~~~~
In file included from /usr/include/glib-2.0/glib/galloca.h:32,
                 from /usr/include/glib-2.0/glib.h:30,
                 from /usr/include/glib-2.0/gobject/gbinding.h:28,
                 from /usr/include/glib-2.0/glib-object.h:22,
                 from /usr/include/glib-2.0/gio/gioenums.h:28,
                 from /usr/include/glib-2.0/gio/giotypes.h:28,
                 from /usr/include/glib-2.0/gio/gio.h:26,
                 from /usr/include/gtk-2.0/gdk/gdkapplaunchcontext.h:30,
                 from /usr/include/gtk-2.0/gdk/gdk.h:32,
                 from /usr/include/gtk-2.0/gtk/gtk.h:32,
                 from src/bimp-gui.c:5:
/usr/include/glib-2.0/glib/gtypes.h:547:8: note: declared here
  547 | struct _GTimeVal
      |        ^~~~~~~~~
In file included from /usr/include/gtk-2.0/gtk/gtkobject.h:37,
                 from /usr/include/gtk-2.0/gtk/gtkwidget.h:36,
                 from /usr/include/gtk-2.0/gtk/gtkcontainer.h:35,
                 from /usr/include/gtk-2.0/gtk/gtkbin.h:35,
                 from /usr/include/gtk-2.0/gtk/gtkwindow.h:36,
                 from /usr/include/gtk-2.0/gtk/gtkdialog.h:35,
                 from /usr/include/gtk-2.0/gtk/gtkaboutdialog.h:32,
                 from /usr/include/gtk-2.0/gtk/gtk.h:33,
                 from src/bimp-manipulations-gui.c:5:
/usr/include/gtk-2.0/gtk/gtktypeutils.h:236:1: warning: 'GTypeDebugFlags' is deprecated [-Wdeprecated-declarations]
  236 | void            gtk_type_init   (GTypeDebugFlags    debug_flags);
      | ^~~~
In file included from /usr/include/glib-2.0/gobject/gobject.h:24,
                 from /usr/include/glib-2.0/gobject/gbinding.h:29,
                 from /usr/include/glib-2.0/glib-object.h:22,
                 from /usr/include/glib-2.0/gio/gioenums.h:28,
                 from /usr/include/glib-2.0/gio/giotypes.h:28,
                 from /usr/include/glib-2.0/gio/gio.h:26,
                 from /usr/include/gtk-2.0/gdk/gdkapplaunchcontext.h:30,
                 from /usr/include/gtk-2.0/gdk/gdk.h:32,
                 from /usr/include/gtk-2.0/gtk/gtk.h:32,
                 from src/bimp-manipulations-gui.c:5:
/usr/include/glib-2.0/gobject/gtype.h:679:1: note: declared here
  679 | {
      | ^
In file included from /usr/include/gtk-2.0/gtk/gtktoolitem.h:31,
                 from /usr/include/gtk-2.0/gtk/gtktoolbutton.h:30,
                 from /usr/include/gtk-2.0/gtk/gtkmenutoolbutton.h:30,
                 from /usr/include/gtk-2.0/gtk/gtk.h:126,
                 from src/bimp-manipulations-gui.c:5:
/usr/include/gtk-2.0/gtk/gtktooltips.h:73:3: warning: 'GTimeVal' is deprecated: Use 'GDateTime' instead [-Wdeprecated-declarations]
   73 |   GTimeVal last_popdown;
      |   ^~~~~~~~
In file included from /usr/include/glib-2.0/glib/galloca.h:32,
                 from /usr/include/glib-2.0/glib.h:30,
                 from /usr/include/glib-2.0/gobject/gbinding.h:28,
                 from /usr/include/glib-2.0/glib-object.h:22,
                 from /usr/include/glib-2.0/gio/gioenums.h:28,
                 from /usr/include/glib-2.0/gio/giotypes.h:28,
                 from /usr/include/glib-2.0/gio/gio.h:26,
                 from /usr/include/gtk-2.0/gdk/gdkapplaunchcontext.h:30,
                 from /usr/include/gtk-2.0/gdk/gdk.h:32,
                 from /usr/include/gtk-2.0/gtk/gtk.h:32,
                 from src/bimp-manipulations-gui.c:5:
/usr/include/glib-2.0/glib/gtypes.h:547:8: note: declared here
  547 | struct _GTimeVal
      |        ^~~~~~~~~
src/bimp-manipulations-gui.c: In function 'bimp_open_editwindow':
src/bimp-manipulations-gui.c:48:14: warning: assignment to 'void (*)(void *)' from incompatible pointer type 'void (*)(struct manip_resize_set *)' [-Wincompatible-pointer-types]
   48 |         save = &bimp_resize_save;
      |              ^
src/bimp-manipulations-gui.c:52:14: warning: assignment to 'void (*)(void *)' from incompatible pointer type 'void (*)(struct manip_crop_set *)' [-Wincompatible-pointer-types]
   52 |         save = &bimp_crop_save;
      |              ^
src/bimp-manipulations-gui.c:56:14: warning: assignment to 'void (*)(void *)' from incompatible pointer type 'void (*)(struct manip_fliprotate_set *)' [-Wincompatible-pointer-types]
   56 |         save = &bimp_fliprotate_save;
      |              ^
src/bimp-manipulations-gui.c:60:14: warning: assignment to 'void (*)(void *)' from incompatible pointer type 'void (*)(struct manip_color_set *)' [-Wincompatible-pointer-types]
   60 |         save = &bimp_color_save;
      |              ^
src/bimp-manipulations-gui.c:64:14: warning: assignment to 'void (*)(void *)' from incompatible pointer type 'void (*)(struct manip_sharpblur_set *)' [-Wincompatible-pointer-types]
   64 |         save = &bimp_sharpblur_save;
      |              ^
src/bimp-manipulations-gui.c:68:14: warning: assignment to 'void (*)(void *)' from incompatible pointer type 'void (*)(struct manip_changeformat_set *)' [-Wincompatible-pointer-types]
   68 |         save = &bimp_changeformat_save;
      |              ^
src/bimp-manipulations-gui.c:72:14: warning: assignment to 'void (*)(void *)' from incompatible pointer type 'void (*)(struct manip_watermark_set *)' [-Wincompatible-pointer-types]
   72 |         save = &bimp_watermark_save;
      |              ^
src/bimp-manipulations-gui.c:76:14: warning: assignment to 'void (*)(void *)' from incompatible pointer type 'void (*)(struct manip_rename_set *)' [-Wincompatible-pointer-types]
   76 |         save = &bimp_rename_save;
      |              ^
src/bimp-manipulations-gui.c:80:14: warning: assignment to 'void (*)(void *)' from incompatible pointer type 'void (*)(struct manip_userdef_set *)' [-Wincompatible-pointer-types]
   80 |         save = &bimp_userdef_save;
      |              ^
In file included from /usr/include/gtk-2.0/gtk/gtkobject.h:37,
                 from /usr/include/gtk-2.0/gtk/gtkwidget.h:36,
                 from /usr/include/gtk-2.0/gtk/gtkcontainer.h:35,
                 from /usr/include/gtk-2.0/gtk/gtkbin.h:35,
                 from /usr/include/gtk-2.0/gtk/gtkwindow.h:36,
                 from /usr/include/gtk-2.0/gtk/gtkdialog.h:35,
                 from /usr/include/gtk-2.0/gtk/gtkaboutdialog.h:32,
                 from /usr/include/gtk-2.0/gtk/gtk.h:33,
                 from src/bimp-manipulations.c:6:
/usr/include/gtk-2.0/gtk/gtktypeutils.h:236:1: warning: 'GTypeDebugFlags' is deprecated [-Wdeprecated-declarations]
  236 | void            gtk_type_init   (GTypeDebugFlags    debug_flags);
      | ^~~~
In file included from /usr/include/glib-2.0/gobject/gobject.h:24,
                 from /usr/include/glib-2.0/gobject/gbinding.h:29,
                 from /usr/include/glib-2.0/glib-object.h:22,
                 from /usr/include/glib-2.0/gio/gioenums.h:28,
                 from /usr/include/glib-2.0/gio/giotypes.h:28,
                 from /usr/include/glib-2.0/gio/gio.h:26,
                 from /usr/include/gtk-2.0/gdk/gdkapplaunchcontext.h:30,
                 from /usr/include/gtk-2.0/gdk/gdk.h:32,
                 from /usr/include/gtk-2.0/gtk/gtk.h:32,
                 from src/bimp-manipulations.c:6:
/usr/include/glib-2.0/gobject/gtype.h:679:1: note: declared here
  679 | {
      | ^
In file included from /usr/include/gtk-2.0/gtk/gtktoolitem.h:31,
                 from /usr/include/gtk-2.0/gtk/gtktoolbutton.h:30,
                 from /usr/include/gtk-2.0/gtk/gtkmenutoolbutton.h:30,
                 from /usr/include/gtk-2.0/gtk/gtk.h:126,
                 from src/bimp-manipulations.c:6:
/usr/include/gtk-2.0/gtk/gtktooltips.h:73:3: warning: 'GTimeVal' is deprecated: Use 'GDateTime' instead [-Wdeprecated-declarations]
   73 |   GTimeVal last_popdown;
      |   ^~~~~~~~
In file included from /usr/include/glib-2.0/glib/galloca.h:32,
                 from /usr/include/glib-2.0/glib.h:30,
                 from /usr/include/glib-2.0/gobject/gbinding.h:28,
                 from /usr/include/glib-2.0/glib-object.h:22,
                 from /usr/include/glib-2.0/gio/gioenums.h:28,
                 from /usr/include/glib-2.0/gio/giotypes.h:28,
                 from /usr/include/glib-2.0/gio/gio.h:26,
                 from /usr/include/gtk-2.0/gdk/gdkapplaunchcontext.h:30,
                 from /usr/include/gtk-2.0/gdk/gdk.h:32,
                 from /usr/include/gtk-2.0/gtk/gtk.h:32,
                 from src/bimp-manipulations.c:6:
/usr/include/glib-2.0/glib/gtypes.h:547:8: note: declared here
  547 | struct _GTimeVal
      |        ^~~~~~~~~
In file included from /usr/include/gtk-2.0/gtk/gtkobject.h:37,
                 from /usr/include/gtk-2.0/gtk/gtkwidget.h:36,
                 from /usr/include/gtk-2.0/gtk/gtkcontainer.h:35,
                 from /usr/include/gtk-2.0/gtk/gtkbin.h:35,
                 from /usr/include/gtk-2.0/gtk/gtkwindow.h:36,
                 from /usr/include/gtk-2.0/gtk/gtkdialog.h:35,
                 from /usr/include/gtk-2.0/gtk/gtkaboutdialog.h:32,
                 from /usr/include/gtk-2.0/gtk/gtk.h:33,
                 from src/bimp-operate.c:6:
/usr/include/gtk-2.0/gtk/gtktypeutils.h:236:1: warning: 'GTypeDebugFlags' is deprecated [-Wdeprecated-declarations]
  236 | void            gtk_type_init   (GTypeDebugFlags    debug_flags);
      | ^~~~
In file included from /usr/include/glib-2.0/gobject/gobject.h:24,
                 from /usr/include/glib-2.0/gobject/gbinding.h:29,
                 from /usr/include/glib-2.0/glib-object.h:22,
                 from /usr/include/glib-2.0/gio/gioenums.h:28,
                 from /usr/include/glib-2.0/gio/giotypes.h:28,
                 from /usr/include/glib-2.0/gio/gio.h:26,
                 from /usr/include/gtk-2.0/gdk/gdkapplaunchcontext.h:30,
                 from /usr/include/gtk-2.0/gdk/gdk.h:32,
                 from /usr/include/gtk-2.0/gtk/gtk.h:32,
                 from src/bimp-operate.c:6:
/usr/include/glib-2.0/gobject/gtype.h:679:1: note: declared here
  679 | {
      | ^
In file included from /usr/include/gtk-2.0/gtk/gtktoolitem.h:31,
                 from /usr/include/gtk-2.0/gtk/gtktoolbutton.h:30,
                 from /usr/include/gtk-2.0/gtk/gtkmenutoolbutton.h:30,
                 from /usr/include/gtk-2.0/gtk/gtk.h:126,
                 from src/bimp-operate.c:6:
/usr/include/gtk-2.0/gtk/gtktooltips.h:73:3: warning: 'GTimeVal' is deprecated: Use 'GDateTime' instead [-Wdeprecated-declarations]
   73 |   GTimeVal last_popdown;
      |   ^~~~~~~~
In file included from /usr/include/glib-2.0/glib/galloca.h:32,
                 from /usr/include/glib-2.0/glib.h:30,
                 from /usr/include/glib-2.0/gobject/gbinding.h:28,
                 from /usr/include/glib-2.0/glib-object.h:22,
                 from /usr/include/glib-2.0/gio/gioenums.h:28,
                 from /usr/include/glib-2.0/gio/giotypes.h:28,
                 from /usr/include/glib-2.0/gio/gio.h:26,
                 from /usr/include/gtk-2.0/gdk/gdkapplaunchcontext.h:30,
                 from /usr/include/gtk-2.0/gdk/gdk.h:32,
                 from /usr/include/gtk-2.0/gtk/gtk.h:32,
                 from src/bimp-operate.c:6:
/usr/include/glib-2.0/glib/gtypes.h:547:8: note: declared here
  547 | struct _GTimeVal
      |        ^~~~~~~~~
src/bimp-operate.c: In function 'process_image':
src/bimp-operate.c:1005:28: warning: '%i' directive writing between 1 and 11 bytes into a region of size 5 [-Wformat-overflow=]
 1005 |         sprintf(strcount, "%i", processed_count + 1);
      |                            ^~
src/bimp-operate.c:1005:27: note: directive argument in the range [-2147483647, 2147483647]
 1005 |         sprintf(strcount, "%i", processed_count + 1);
      |                           ^~~~
src/bimp-operate.c:1005:9: note: 'sprintf' output between 2 and 12 bytes into a destination of size 5
 1005 |         sprintf(strcount, "%i", processed_count + 1);
      |         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /usr/include/gtk-2.0/gtk/gtkobject.h:37,
                 from /usr/include/gtk-2.0/gtk/gtkwidget.h:36,
                 from /usr/include/gtk-2.0/gtk/gtkcontainer.h:35,
                 from /usr/include/gtk-2.0/gtk/gtkbin.h:35,
                 from /usr/include/gtk-2.0/gtk/gtkwindow.h:36,
                 from /usr/include/gtk-2.0/gtk/gtkdialog.h:35,
                 from /usr/include/gtk-2.0/gtk/gtkaboutdialog.h:32,
                 from /usr/include/gtk-2.0/gtk/gtk.h:33,
                 from src/bimp-serialize.c:8:
/usr/include/gtk-2.0/gtk/gtktypeutils.h:236:1: warning: 'GTypeDebugFlags' is deprecated [-Wdeprecated-declarations]
  236 | void            gtk_type_init   (GTypeDebugFlags    debug_flags);
      | ^~~~
In file included from /usr/include/glib-2.0/gobject/gobject.h:24,
                 from /usr/include/glib-2.0/gobject/gbinding.h:29,
                 from /usr/include/glib-2.0/glib-object.h:22,
                 from /usr/include/glib-2.0/gio/gioenums.h:28,
                 from /usr/include/glib-2.0/gio/giotypes.h:28,
                 from /usr/include/glib-2.0/gio/gio.h:26,
                 from /usr/include/gtk-2.0/gdk/gdkapplaunchcontext.h:30,
                 from /usr/include/gtk-2.0/gdk/gdk.h:32,
                 from /usr/include/gtk-2.0/gtk/gtk.h:32,
                 from src/bimp-serialize.c:8:
/usr/include/glib-2.0/gobject/gtype.h:679:1: note: declared here
  679 | {
      | ^
In file included from /usr/include/gtk-2.0/gtk/gtktoolitem.h:31,
                 from /usr/include/gtk-2.0/gtk/gtktoolbutton.h:30,
                 from /usr/include/gtk-2.0/gtk/gtkmenutoolbutton.h:30,
                 from /usr/include/gtk-2.0/gtk/gtk.h:126,
                 from src/bimp-serialize.c:8:
/usr/include/gtk-2.0/gtk/gtktooltips.h:73:3: warning: 'GTimeVal' is deprecated: Use 'GDateTime' instead [-Wdeprecated-declarations]
   73 |   GTimeVal last_popdown;
      |   ^~~~~~~~
In file included from /usr/include/glib-2.0/glib/galloca.h:32,
                 from /usr/include/glib-2.0/glib.h:30,
                 from src/bimp-serialize.c:6:
/usr/include/glib-2.0/glib/gtypes.h:547:8: note: declared here
  547 | struct _GTimeVal
      |        ^~~~~~~~~
In file included from /usr/include/gtk-2.0/gtk/gtkobject.h:37,
                 from /usr/include/gtk-2.0/gtk/gtkwidget.h:36,
                 from /usr/include/gtk-2.0/gtk/gtkcontainer.h:35,
                 from /usr/include/gtk-2.0/gtk/gtkbin.h:35,
                 from /usr/include/gtk-2.0/gtk/gtkwindow.h:36,
                 from /usr/include/gtk-2.0/gtk/gtkdialog.h:35,
                 from /usr/include/gtk-2.0/gtk/gtkaboutdialog.h:32,
                 from /usr/include/gtk-2.0/gtk/gtk.h:33,
                 from src/bimp-utils.c:7:
/usr/include/gtk-2.0/gtk/gtktypeutils.h:236:1: warning: 'GTypeDebugFlags' is deprecated [-Wdeprecated-declarations]
  236 | void            gtk_type_init   (GTypeDebugFlags    debug_flags);
      | ^~~~
In file included from /usr/include/glib-2.0/gobject/gobject.h:24,
                 from /usr/include/glib-2.0/gobject/gbinding.h:29,
                 from /usr/include/glib-2.0/glib-object.h:22,
                 from /usr/include/glib-2.0/gio/gioenums.h:28,
                 from /usr/include/glib-2.0/gio/giotypes.h:28,
                 from /usr/include/glib-2.0/gio/gio.h:26,
                 from /usr/include/gtk-2.0/gdk/gdkapplaunchcontext.h:30,
                 from /usr/include/gtk-2.0/gdk/gdk.h:32,
                 from /usr/include/gtk-2.0/gtk/gtk.h:32,
                 from src/bimp-utils.c:7:
/usr/include/glib-2.0/gobject/gtype.h:679:1: note: declared here
  679 | {
      | ^
In file included from /usr/include/gtk-2.0/gtk/gtktoolitem.h:31,
                 from /usr/include/gtk-2.0/gtk/gtktoolbutton.h:30,
                 from /usr/include/gtk-2.0/gtk/gtkmenutoolbutton.h:30,
                 from /usr/include/gtk-2.0/gtk/gtk.h:126,
                 from src/bimp-utils.c:7:
/usr/include/gtk-2.0/gtk/gtktooltips.h:73:3: warning: 'GTimeVal' is deprecated: Use 'GDateTime' instead [-Wdeprecated-declarations]
   73 |   GTimeVal last_popdown;
      |   ^~~~~~~~
In file included from /usr/include/glib-2.0/glib/galloca.h:32,
                 from /usr/include/glib-2.0/glib.h:30,
                 from /usr/include/glib-2.0/gobject/gbinding.h:28,
                 from /usr/include/glib-2.0/glib-object.h:22,
                 from /usr/include/glib-2.0/gio/gioenums.h:28,
                 from /usr/include/glib-2.0/gio/giotypes.h:28,
                 from /usr/include/glib-2.0/gio/gio.h:26,
                 from /usr/include/gtk-2.0/gdk/gdkapplaunchcontext.h:30,
                 from /usr/include/gtk-2.0/gdk/gdk.h:32,
                 from /usr/include/gtk-2.0/gtk/gtk.h:32,
                 from src/bimp-utils.c:7:
/usr/include/glib-2.0/glib/gtypes.h:547:8: note: declared here
  547 | struct _GTimeVal
      |        ^~~~~~~~~
In file included from /usr/include/gtk-2.0/gtk/gtkobject.h:37,
                 from /usr/include/gtk-2.0/gtk/gtkwidget.h:36,
                 from /usr/include/gtk-2.0/gtk/gtkcontainer.h:35,
                 from /usr/include/gtk-2.0/gtk/gtkbin.h:35,
                 from /usr/include/gtk-2.0/gtk/gtkwindow.h:36,
                 from /usr/include/gtk-2.0/gtk/gtkdialog.h:35,
                 from /usr/include/gtk-2.0/gtk/gtkaboutdialog.h:32,
                 from /usr/include/gtk-2.0/gtk/gtk.h:33,
                 from src/bimp.c:26:
/usr/include/gtk-2.0/gtk/gtktypeutils.h:236:1: warning: 'GTypeDebugFlags' is deprecated [-Wdeprecated-declarations]
  236 | void            gtk_type_init   (GTypeDebugFlags    debug_flags);
      | ^~~~
In file included from /usr/include/glib-2.0/gobject/gobject.h:24,
                 from /usr/include/glib-2.0/gobject/gbinding.h:29,
                 from /usr/include/glib-2.0/glib-object.h:22,
                 from /usr/include/gegl-0.4/gegl.h:22,
                 from /usr/include/gimp-2.0/libgimp/gimp.h:25,
                 from src/bimp.c:25:
/usr/include/glib-2.0/gobject/gtype.h:679:1: note: declared here
  679 | {
      | ^
In file included from /usr/include/gtk-2.0/gtk/gtktoolitem.h:31,
                 from /usr/include/gtk-2.0/gtk/gtktoolbutton.h:30,
                 from /usr/include/gtk-2.0/gtk/gtkmenutoolbutton.h:30,
                 from /usr/include/gtk-2.0/gtk/gtk.h:126,
                 from src/bimp.c:26:
/usr/include/gtk-2.0/gtk/gtktooltips.h:73:3: warning: 'GTimeVal' is deprecated: Use 'GDateTime' instead [-Wdeprecated-declarations]
   73 |   GTimeVal last_popdown;
      |   ^~~~~~~~
In file included from /usr/include/glib-2.0/glib/galloca.h:32,
                 from /usr/include/glib-2.0/glib.h:30,
                 from /usr/include/glib-2.0/gobject/gbinding.h:28,
                 from /usr/include/glib-2.0/glib-object.h:22,
                 from /usr/include/gegl-0.4/gegl.h:22,
                 from /usr/include/gimp-2.0/libgimp/gimp.h:25,
                 from src/bimp.c:25:
/usr/include/glib-2.0/glib/gtypes.h:547:8: note: declared here
  547 | struct _GTimeVal
      |        ^~~~~~~~~
In file included from /usr/include/gtk-2.0/gtk/gtkobject.h:37,
                 from /usr/include/gtk-2.0/gtk/gtkwidget.h:36,
                 from /usr/include/gtk-2.0/gtk/gtkcontainer.h:35,
                 from /usr/include/gtk-2.0/gtk/gtkbin.h:35,
                 from /usr/include/gtk-2.0/gtk/gtkwindow.h:36,
                 from /usr/include/gtk-2.0/gtk/gtkdialog.h:35,
                 from /usr/include/gtk-2.0/gtk/gtkaboutdialog.h:32,
                 from /usr/include/gtk-2.0/gtk/gtk.h:33,
                 from src/manipulation-gui/gui-changeformat.c:1:
/usr/include/gtk-2.0/gtk/gtktypeutils.h:236:1: warning: 'GTypeDebugFlags' is deprecated [-Wdeprecated-declarations]
  236 | void            gtk_type_init   (GTypeDebugFlags    debug_flags);
      | ^~~~
In file included from /usr/include/glib-2.0/gobject/gobject.h:24,
                 from /usr/include/glib-2.0/gobject/gbinding.h:29,
                 from /usr/include/glib-2.0/glib-object.h:22,
                 from /usr/include/glib-2.0/gio/gioenums.h:28,
                 from /usr/include/glib-2.0/gio/giotypes.h:28,
                 from /usr/include/glib-2.0/gio/gio.h:26,
                 from /usr/include/gtk-2.0/gdk/gdkapplaunchcontext.h:30,
                 from /usr/include/gtk-2.0/gdk/gdk.h:32,
                 from /usr/include/gtk-2.0/gtk/gtk.h:32,
                 from src/manipulation-gui/gui-changeformat.c:1:
/usr/include/glib-2.0/gobject/gtype.h:679:1: note: declared here
  679 | {
      | ^
In file included from /usr/include/gtk-2.0/gtk/gtktoolitem.h:31,
                 from /usr/include/gtk-2.0/gtk/gtktoolbutton.h:30,
                 from /usr/include/gtk-2.0/gtk/gtkmenutoolbutton.h:30,
                 from /usr/include/gtk-2.0/gtk/gtk.h:126,
                 from src/manipulation-gui/gui-changeformat.c:1:
/usr/include/gtk-2.0/gtk/gtktooltips.h:73:3: warning: 'GTimeVal' is deprecated: Use 'GDateTime' instead [-Wdeprecated-declarations]
   73 |   GTimeVal last_popdown;
      |   ^~~~~~~~
In file included from /usr/include/glib-2.0/glib/galloca.h:32,
                 from /usr/include/glib-2.0/glib.h:30,
                 from /usr/include/glib-2.0/gobject/gbinding.h:28,
                 from /usr/include/glib-2.0/glib-object.h:22,
                 from /usr/include/glib-2.0/gio/gioenums.h:28,
                 from /usr/include/glib-2.0/gio/giotypes.h:28,
                 from /usr/include/glib-2.0/gio/gio.h:26,
                 from /usr/include/gtk-2.0/gdk/gdkapplaunchcontext.h:30,
                 from /usr/include/gtk-2.0/gdk/gdk.h:32,
                 from /usr/include/gtk-2.0/gtk/gtk.h:32,
                 from src/manipulation-gui/gui-changeformat.c:1:
/usr/include/glib-2.0/glib/gtypes.h:547:8: note: declared here
  547 | struct _GTimeVal
      |        ^~~~~~~~~
In file included from /usr/include/gtk-2.0/gtk/gtkobject.h:37,
                 from /usr/include/gtk-2.0/gtk/gtkwidget.h:36,
                 from /usr/include/gtk-2.0/gtk/gtkcontainer.h:35,
                 from /usr/include/gtk-2.0/gtk/gtkbin.h:35,
                 from /usr/include/gtk-2.0/gtk/gtkwindow.h:36,
                 from /usr/include/gtk-2.0/gtk/gtkdialog.h:35,
                 from /usr/include/gtk-2.0/gtk/gtkaboutdialog.h:32,
                 from /usr/include/gtk-2.0/gtk/gtk.h:33,
                 from src/manipulation-gui/gui-color.c:1:
/usr/include/gtk-2.0/gtk/gtktypeutils.h:236:1: warning: 'GTypeDebugFlags' is deprecated [-Wdeprecated-declarations]
  236 | void            gtk_type_init   (GTypeDebugFlags    debug_flags);
      | ^~~~
In file included from /usr/include/glib-2.0/gobject/gobject.h:24,
                 from /usr/include/glib-2.0/gobject/gbinding.h:29,
                 from /usr/include/glib-2.0/glib-object.h:22,
                 from /usr/include/glib-2.0/gio/gioenums.h:28,
                 from /usr/include/glib-2.0/gio/giotypes.h:28,
                 from /usr/include/glib-2.0/gio/gio.h:26,
                 from /usr/include/gtk-2.0/gdk/gdkapplaunchcontext.h:30,
                 from /usr/include/gtk-2.0/gdk/gdk.h:32,
                 from /usr/include/gtk-2.0/gtk/gtk.h:32,
                 from src/manipulation-gui/gui-color.c:1:
/usr/include/glib-2.0/gobject/gtype.h:679:1: note: declared here
  679 | {
      | ^
In file included from /usr/include/gtk-2.0/gtk/gtktoolitem.h:31,
                 from /usr/include/gtk-2.0/gtk/gtktoolbutton.h:30,
                 from /usr/include/gtk-2.0/gtk/gtkmenutoolbutton.h:30,
                 from /usr/include/gtk-2.0/gtk/gtk.h:126,
                 from src/manipulation-gui/gui-color.c:1:
/usr/include/gtk-2.0/gtk/gtktooltips.h:73:3: warning: 'GTimeVal' is deprecated: Use 'GDateTime' instead [-Wdeprecated-declarations]
   73 |   GTimeVal last_popdown;
      |   ^~~~~~~~
In file included from /usr/include/glib-2.0/glib/galloca.h:32,
                 from /usr/include/glib-2.0/glib.h:30,
                 from /usr/include/glib-2.0/gobject/gbinding.h:28,
                 from /usr/include/glib-2.0/glib-object.h:22,
                 from /usr/include/glib-2.0/gio/gioenums.h:28,
                 from /usr/include/glib-2.0/gio/giotypes.h:28,
                 from /usr/include/glib-2.0/gio/gio.h:26,
                 from /usr/include/gtk-2.0/gdk/gdkapplaunchcontext.h:30,
                 from /usr/include/gtk-2.0/gdk/gdk.h:32,
                 from /usr/include/gtk-2.0/gtk/gtk.h:32,
                 from src/manipulation-gui/gui-color.c:1:
/usr/include/glib-2.0/glib/gtypes.h:547:8: note: declared here
  547 | struct _GTimeVal
      |        ^~~~~~~~~
In file included from /usr/include/gtk-2.0/gtk/gtkobject.h:37,
                 from /usr/include/gtk-2.0/gtk/gtkwidget.h:36,
                 from /usr/include/gtk-2.0/gtk/gtkcontainer.h:35,
                 from /usr/include/gtk-2.0/gtk/gtkbin.h:35,
                 from /usr/include/gtk-2.0/gtk/gtkwindow.h:36,
                 from /usr/include/gtk-2.0/gtk/gtkdialog.h:35,
                 from /usr/include/gtk-2.0/gtk/gtkaboutdialog.h:32,
                 from /usr/include/gtk-2.0/gtk/gtk.h:33,
                 from src/manipulation-gui/gui-crop.c:1:
/usr/include/gtk-2.0/gtk/gtktypeutils.h:236:1: warning: 'GTypeDebugFlags' is deprecated [-Wdeprecated-declarations]
  236 | void            gtk_type_init   (GTypeDebugFlags    debug_flags);
      | ^~~~
In file included from /usr/include/glib-2.0/gobject/gobject.h:24,
                 from /usr/include/glib-2.0/gobject/gbinding.h:29,
                 from /usr/include/glib-2.0/glib-object.h:22,
                 from /usr/include/glib-2.0/gio/gioenums.h:28,
                 from /usr/include/glib-2.0/gio/giotypes.h:28,
                 from /usr/include/glib-2.0/gio/gio.h:26,
                 from /usr/include/gtk-2.0/gdk/gdkapplaunchcontext.h:30,
                 from /usr/include/gtk-2.0/gdk/gdk.h:32,
                 from /usr/include/gtk-2.0/gtk/gtk.h:32,
                 from src/manipulation-gui/gui-crop.c:1:
/usr/include/glib-2.0/gobject/gtype.h:679:1: note: declared here
  679 | {
      | ^
In file included from /usr/include/gtk-2.0/gtk/gtktoolitem.h:31,
                 from /usr/include/gtk-2.0/gtk/gtktoolbutton.h:30,
                 from /usr/include/gtk-2.0/gtk/gtkmenutoolbutton.h:30,
                 from /usr/include/gtk-2.0/gtk/gtk.h:126,
                 from src/manipulation-gui/gui-crop.c:1:
/usr/include/gtk-2.0/gtk/gtktooltips.h:73:3: warning: 'GTimeVal' is deprecated: Use 'GDateTime' instead [-Wdeprecated-declarations]
   73 |   GTimeVal last_popdown;
      |   ^~~~~~~~
In file included from /usr/include/glib-2.0/glib/galloca.h:32,
                 from /usr/include/glib-2.0/glib.h:30,
                 from /usr/include/glib-2.0/gobject/gbinding.h:28,
                 from /usr/include/glib-2.0/glib-object.h:22,
                 from /usr/include/glib-2.0/gio/gioenums.h:28,
                 from /usr/include/glib-2.0/gio/giotypes.h:28,
                 from /usr/include/glib-2.0/gio/gio.h:26,
                 from /usr/include/gtk-2.0/gdk/gdkapplaunchcontext.h:30,
                 from /usr/include/gtk-2.0/gdk/gdk.h:32,
                 from /usr/include/gtk-2.0/gtk/gtk.h:32,
                 from src/manipulation-gui/gui-crop.c:1:
/usr/include/glib-2.0/glib/gtypes.h:547:8: note: declared here
  547 | struct _GTimeVal
      |        ^~~~~~~~~
In file included from /usr/include/gtk-2.0/gtk/gtkobject.h:37,
                 from /usr/include/gtk-2.0/gtk/gtkwidget.h:36,
                 from /usr/include/gtk-2.0/gtk/gtkcontainer.h:35,
                 from /usr/include/gtk-2.0/gtk/gtkbin.h:35,
                 from /usr/include/gtk-2.0/gtk/gtkwindow.h:36,
                 from /usr/include/gtk-2.0/gtk/gtkdialog.h:35,
                 from /usr/include/gtk-2.0/gtk/gtkaboutdialog.h:32,
                 from /usr/include/gtk-2.0/gtk/gtk.h:33,
                 from src/manipulation-gui/gui-fliprotate.c:1:
/usr/include/gtk-2.0/gtk/gtktypeutils.h:236:1: warning: 'GTypeDebugFlags' is deprecated [-Wdeprecated-declarations]
  236 | void            gtk_type_init   (GTypeDebugFlags    debug_flags);
      | ^~~~
In file included from /usr/include/glib-2.0/gobject/gobject.h:24,
                 from /usr/include/glib-2.0/gobject/gbinding.h:29,
                 from /usr/include/glib-2.0/glib-object.h:22,
                 from /usr/include/glib-2.0/gio/gioenums.h:28,
                 from /usr/include/glib-2.0/gio/giotypes.h:28,
                 from /usr/include/glib-2.0/gio/gio.h:26,
                 from /usr/include/gtk-2.0/gdk/gdkapplaunchcontext.h:30,
                 from /usr/include/gtk-2.0/gdk/gdk.h:32,
                 from /usr/include/gtk-2.0/gtk/gtk.h:32,
                 from src/manipulation-gui/gui-fliprotate.c:1:
/usr/include/glib-2.0/gobject/gtype.h:679:1: note: declared here
  679 | {
      | ^
In file included from /usr/include/gtk-2.0/gtk/gtktoolitem.h:31,
                 from /usr/include/gtk-2.0/gtk/gtktoolbutton.h:30,
                 from /usr/include/gtk-2.0/gtk/gtkmenutoolbutton.h:30,
                 from /usr/include/gtk-2.0/gtk/gtk.h:126,
                 from src/manipulation-gui/gui-fliprotate.c:1:
/usr/include/gtk-2.0/gtk/gtktooltips.h:73:3: warning: 'GTimeVal' is deprecated: Use 'GDateTime' instead [-Wdeprecated-declarations]
   73 |   GTimeVal last_popdown;
      |   ^~~~~~~~
In file included from /usr/include/glib-2.0/glib/galloca.h:32,
                 from /usr/include/glib-2.0/glib.h:30,
                 from /usr/include/glib-2.0/gobject/gbinding.h:28,
                 from /usr/include/glib-2.0/glib-object.h:22,
                 from /usr/include/glib-2.0/gio/gioenums.h:28,
                 from /usr/include/glib-2.0/gio/giotypes.h:28,
                 from /usr/include/glib-2.0/gio/gio.h:26,
                 from /usr/include/gtk-2.0/gdk/gdkapplaunchcontext.h:30,
                 from /usr/include/gtk-2.0/gdk/gdk.h:32,
                 from /usr/include/gtk-2.0/gtk/gtk.h:32,
                 from src/manipulation-gui/gui-fliprotate.c:1:
/usr/include/glib-2.0/glib/gtypes.h:547:8: note: declared here
  547 | struct _GTimeVal
      |        ^~~~~~~~~
In file included from /usr/include/gtk-2.0/gtk/gtkobject.h:37,
                 from /usr/include/gtk-2.0/gtk/gtkwidget.h:36,
                 from /usr/include/gtk-2.0/gtk/gtkcontainer.h:35,
                 from /usr/include/gtk-2.0/gtk/gtkbin.h:35,
                 from /usr/include/gtk-2.0/gtk/gtkwindow.h:36,
                 from /usr/include/gtk-2.0/gtk/gtkdialog.h:35,
                 from /usr/include/gtk-2.0/gtk/gtkaboutdialog.h:32,
                 from /usr/include/gtk-2.0/gtk/gtk.h:33,
                 from src/manipulation-gui/gui-rename.c:3:
/usr/include/gtk-2.0/gtk/gtktypeutils.h:236:1: warning: 'GTypeDebugFlags' is deprecated [-Wdeprecated-declarations]
  236 | void            gtk_type_init   (GTypeDebugFlags    debug_flags);
      | ^~~~
In file included from /usr/include/glib-2.0/gobject/gobject.h:24,
                 from /usr/include/glib-2.0/gobject/gbinding.h:29,
                 from /usr/include/glib-2.0/glib-object.h:22,
                 from /usr/include/glib-2.0/gio/gioenums.h:28,
                 from /usr/include/glib-2.0/gio/giotypes.h:28,
                 from /usr/include/glib-2.0/gio/gio.h:26,
                 from /usr/include/gtk-2.0/gdk/gdkapplaunchcontext.h:30,
                 from /usr/include/gtk-2.0/gdk/gdk.h:32,
                 from /usr/include/gtk-2.0/gtk/gtk.h:32,
                 from src/manipulation-gui/gui-rename.c:3:
/usr/include/glib-2.0/gobject/gtype.h:679:1: note: declared here
  679 | {
      | ^
In file included from /usr/include/gtk-2.0/gtk/gtktoolitem.h:31,
                 from /usr/include/gtk-2.0/gtk/gtktoolbutton.h:30,
                 from /usr/include/gtk-2.0/gtk/gtkmenutoolbutton.h:30,
                 from /usr/include/gtk-2.0/gtk/gtk.h:126,
                 from src/manipulation-gui/gui-rename.c:3:
/usr/include/gtk-2.0/gtk/gtktooltips.h:73:3: warning: 'GTimeVal' is deprecated: Use 'GDateTime' instead [-Wdeprecated-declarations]
   73 |   GTimeVal last_popdown;
      |   ^~~~~~~~
In file included from /usr/include/glib-2.0/glib/galloca.h:32,
                 from /usr/include/glib-2.0/glib.h:30,
                 from /usr/include/glib-2.0/gobject/gbinding.h:28,
                 from /usr/include/glib-2.0/glib-object.h:22,
                 from /usr/include/glib-2.0/gio/gioenums.h:28,
                 from /usr/include/glib-2.0/gio/giotypes.h:28,
                 from /usr/include/glib-2.0/gio/gio.h:26,
                 from /usr/include/gtk-2.0/gdk/gdkapplaunchcontext.h:30,
                 from /usr/include/gtk-2.0/gdk/gdk.h:32,
                 from /usr/include/gtk-2.0/gtk/gtk.h:32,
                 from src/manipulation-gui/gui-rename.c:3:
/usr/include/glib-2.0/glib/gtypes.h:547:8: note: declared here
  547 | struct _GTimeVal
      |        ^~~~~~~~~
In file included from /usr/include/gtk-2.0/gtk/gtkobject.h:37,
                 from /usr/include/gtk-2.0/gtk/gtkwidget.h:36,
                 from /usr/include/gtk-2.0/gtk/gtkcontainer.h:35,
                 from /usr/include/gtk-2.0/gtk/gtkbin.h:35,
                 from /usr/include/gtk-2.0/gtk/gtkwindow.h:36,
                 from /usr/include/gtk-2.0/gtk/gtkdialog.h:35,
                 from /usr/include/gtk-2.0/gtk/gtkaboutdialog.h:32,
                 from /usr/include/gtk-2.0/gtk/gtk.h:33,
                 from src/manipulation-gui/gui-resize.c:1:
/usr/include/gtk-2.0/gtk/gtktypeutils.h:236:1: warning: 'GTypeDebugFlags' is deprecated [-Wdeprecated-declarations]
  236 | void            gtk_type_init   (GTypeDebugFlags    debug_flags);
      | ^~~~
In file included from /usr/include/glib-2.0/gobject/gobject.h:24,
                 from /usr/include/glib-2.0/gobject/gbinding.h:29,
                 from /usr/include/glib-2.0/glib-object.h:22,
                 from /usr/include/glib-2.0/gio/gioenums.h:28,
                 from /usr/include/glib-2.0/gio/giotypes.h:28,
                 from /usr/include/glib-2.0/gio/gio.h:26,
                 from /usr/include/gtk-2.0/gdk/gdkapplaunchcontext.h:30,
                 from /usr/include/gtk-2.0/gdk/gdk.h:32,
                 from /usr/include/gtk-2.0/gtk/gtk.h:32,
                 from src/manipulation-gui/gui-resize.c:1:
/usr/include/glib-2.0/gobject/gtype.h:679:1: note: declared here
  679 | {
      | ^
In file included from /usr/include/gtk-2.0/gtk/gtktoolitem.h:31,
                 from /usr/include/gtk-2.0/gtk/gtktoolbutton.h:30,
                 from /usr/include/gtk-2.0/gtk/gtkmenutoolbutton.h:30,
                 from /usr/include/gtk-2.0/gtk/gtk.h:126,
                 from src/manipulation-gui/gui-resize.c:1:
/usr/include/gtk-2.0/gtk/gtktooltips.h:73:3: warning: 'GTimeVal' is deprecated: Use 'GDateTime' instead [-Wdeprecated-declarations]
   73 |   GTimeVal last_popdown;
      |   ^~~~~~~~
In file included from /usr/include/glib-2.0/glib/galloca.h:32,
                 from /usr/include/glib-2.0/glib.h:30,
                 from /usr/include/glib-2.0/gobject/gbinding.h:28,
                 from /usr/include/glib-2.0/glib-object.h:22,
                 from /usr/include/glib-2.0/gio/gioenums.h:28,
                 from /usr/include/glib-2.0/gio/giotypes.h:28,
                 from /usr/include/glib-2.0/gio/gio.h:26,
                 from /usr/include/gtk-2.0/gdk/gdkapplaunchcontext.h:30,
                 from /usr/include/gtk-2.0/gdk/gdk.h:32,
                 from /usr/include/gtk-2.0/gtk/gtk.h:32,
                 from src/manipulation-gui/gui-resize.c:1:
/usr/include/glib-2.0/glib/gtypes.h:547:8: note: declared here
  547 | struct _GTimeVal
      |        ^~~~~~~~~
In file included from /usr/include/gtk-2.0/gtk/gtkobject.h:37,
                 from /usr/include/gtk-2.0/gtk/gtkwidget.h:36,
                 from /usr/include/gtk-2.0/gtk/gtkcontainer.h:35,
                 from /usr/include/gtk-2.0/gtk/gtkbin.h:35,
                 from /usr/include/gtk-2.0/gtk/gtkwindow.h:36,
                 from /usr/include/gtk-2.0/gtk/gtkdialog.h:35,
                 from /usr/include/gtk-2.0/gtk/gtkaboutdialog.h:32,
                 from /usr/include/gtk-2.0/gtk/gtk.h:33,
                 from src/manipulation-gui/gui-sharpblur.c:1:
/usr/include/gtk-2.0/gtk/gtktypeutils.h:236:1: warning: 'GTypeDebugFlags' is deprecated [-Wdeprecated-declarations]
  236 | void            gtk_type_init   (GTypeDebugFlags    debug_flags);
      | ^~~~
In file included from /usr/include/glib-2.0/gobject/gobject.h:24,
                 from /usr/include/glib-2.0/gobject/gbinding.h:29,
                 from /usr/include/glib-2.0/glib-object.h:22,
                 from /usr/include/glib-2.0/gio/gioenums.h:28,
                 from /usr/include/glib-2.0/gio/giotypes.h:28,
                 from /usr/include/glib-2.0/gio/gio.h:26,
                 from /usr/include/gtk-2.0/gdk/gdkapplaunchcontext.h:30,
                 from /usr/include/gtk-2.0/gdk/gdk.h:32,
                 from /usr/include/gtk-2.0/gtk/gtk.h:32,
                 from src/manipulation-gui/gui-sharpblur.c:1:
/usr/include/glib-2.0/gobject/gtype.h:679:1: note: declared here
  679 | {
      | ^
In file included from /usr/include/gtk-2.0/gtk/gtktoolitem.h:31,
                 from /usr/include/gtk-2.0/gtk/gtktoolbutton.h:30,
                 from /usr/include/gtk-2.0/gtk/gtkmenutoolbutton.h:30,
                 from /usr/include/gtk-2.0/gtk/gtk.h:126,
                 from src/manipulation-gui/gui-sharpblur.c:1:
/usr/include/gtk-2.0/gtk/gtktooltips.h:73:3: warning: 'GTimeVal' is deprecated: Use 'GDateTime' instead [-Wdeprecated-declarations]
   73 |   GTimeVal last_popdown;
      |   ^~~~~~~~
In file included from /usr/include/glib-2.0/glib/galloca.h:32,
                 from /usr/include/glib-2.0/glib.h:30,
                 from /usr/include/glib-2.0/gobject/gbinding.h:28,
                 from /usr/include/glib-2.0/glib-object.h:22,
                 from /usr/include/glib-2.0/gio/gioenums.h:28,
                 from /usr/include/glib-2.0/gio/giotypes.h:28,
                 from /usr/include/glib-2.0/gio/gio.h:26,
                 from /usr/include/gtk-2.0/gdk/gdkapplaunchcontext.h:30,
                 from /usr/include/gtk-2.0/gdk/gdk.h:32,
                 from /usr/include/gtk-2.0/gtk/gtk.h:32,
                 from src/manipulation-gui/gui-sharpblur.c:1:
/usr/include/glib-2.0/glib/gtypes.h:547:8: note: declared here
  547 | struct _GTimeVal
      |        ^~~~~~~~~
In file included from /usr/include/gtk-2.0/gtk/gtkobject.h:37,
                 from /usr/include/gtk-2.0/gtk/gtkwidget.h:36,
                 from /usr/include/gtk-2.0/gtk/gtkcontainer.h:35,
                 from /usr/include/gtk-2.0/gtk/gtkbin.h:35,
                 from /usr/include/gtk-2.0/gtk/gtkwindow.h:36,
                 from /usr/include/gtk-2.0/gtk/gtkdialog.h:35,
                 from /usr/include/gtk-2.0/gtk/gtkaboutdialog.h:32,
                 from /usr/include/gtk-2.0/gtk/gtk.h:33,
                 from src/manipulation-gui/gui-userdef.c:1:
/usr/include/gtk-2.0/gtk/gtktypeutils.h:236:1: warning: 'GTypeDebugFlags' is deprecated [-Wdeprecated-declarations]
  236 | void            gtk_type_init   (GTypeDebugFlags    debug_flags);
      | ^~~~
In file included from /usr/include/glib-2.0/gobject/gobject.h:24,
                 from /usr/include/glib-2.0/gobject/gbinding.h:29,
                 from /usr/include/glib-2.0/glib-object.h:22,
                 from /usr/include/glib-2.0/gio/gioenums.h:28,
                 from /usr/include/glib-2.0/gio/giotypes.h:28,
                 from /usr/include/glib-2.0/gio/gio.h:26,
                 from /usr/include/gtk-2.0/gdk/gdkapplaunchcontext.h:30,
                 from /usr/include/gtk-2.0/gdk/gdk.h:32,
                 from /usr/include/gtk-2.0/gtk/gtk.h:32,
                 from src/manipulation-gui/gui-userdef.c:1:
/usr/include/glib-2.0/gobject/gtype.h:679:1: note: declared here
  679 | {
      | ^
In file included from /usr/include/gtk-2.0/gtk/gtktoolitem.h:31,
                 from /usr/include/gtk-2.0/gtk/gtktoolbutton.h:30,
                 from /usr/include/gtk-2.0/gtk/gtkmenutoolbutton.h:30,
                 from /usr/include/gtk-2.0/gtk/gtk.h:126,
                 from src/manipulation-gui/gui-userdef.c:1:
/usr/include/gtk-2.0/gtk/gtktooltips.h:73:3: warning: 'GTimeVal' is deprecated: Use 'GDateTime' instead [-Wdeprecated-declarations]
   73 |   GTimeVal last_popdown;
      |   ^~~~~~~~
In file included from /usr/include/glib-2.0/glib/galloca.h:32,
                 from /usr/include/glib-2.0/glib.h:30,
                 from /usr/include/glib-2.0/gobject/gbinding.h:28,
                 from /usr/include/glib-2.0/glib-object.h:22,
                 from /usr/include/glib-2.0/gio/gioenums.h:28,
                 from /usr/include/glib-2.0/gio/giotypes.h:28,
                 from /usr/include/glib-2.0/gio/gio.h:26,
                 from /usr/include/gtk-2.0/gdk/gdkapplaunchcontext.h:30,
                 from /usr/include/gtk-2.0/gdk/gdk.h:32,
                 from /usr/include/gtk-2.0/gtk/gtk.h:32,
                 from src/manipulation-gui/gui-userdef.c:1:
/usr/include/glib-2.0/glib/gtypes.h:547:8: note: declared here
  547 | struct _GTimeVal
      |        ^~~~~~~~~
In file included from /usr/include/gtk-2.0/gtk/gtkobject.h:37,
                 from /usr/include/gtk-2.0/gtk/gtkwidget.h:36,
                 from /usr/include/gtk-2.0/gtk/gtkcontainer.h:35,
                 from /usr/include/gtk-2.0/gtk/gtkbin.h:35,
                 from /usr/include/gtk-2.0/gtk/gtkwindow.h:36,
                 from /usr/include/gtk-2.0/gtk/gtkdialog.h:35,
                 from /usr/include/gtk-2.0/gtk/gtkaboutdialog.h:32,
                 from /usr/include/gtk-2.0/gtk/gtk.h:33,
                 from src/manipulation-gui/gui-watermark.c:3:
/usr/include/gtk-2.0/gtk/gtktypeutils.h:236:1: warning: 'GTypeDebugFlags' is deprecated [-Wdeprecated-declarations]
  236 | void            gtk_type_init   (GTypeDebugFlags    debug_flags);
      | ^~~~
In file included from /usr/include/glib-2.0/gobject/gobject.h:24,
                 from /usr/include/glib-2.0/gobject/gbinding.h:29,
                 from /usr/include/glib-2.0/glib-object.h:22,
                 from /usr/include/glib-2.0/gio/gioenums.h:28,
                 from /usr/include/glib-2.0/gio/giotypes.h:28,
                 from /usr/include/glib-2.0/gio/gio.h:26,
                 from /usr/include/gtk-2.0/gdk/gdkapplaunchcontext.h:30,
                 from /usr/include/gtk-2.0/gdk/gdk.h:32,
                 from /usr/include/gtk-2.0/gtk/gtk.h:32,
                 from src/manipulation-gui/gui-watermark.c:3:
/usr/include/glib-2.0/gobject/gtype.h:679:1: note: declared here
  679 | {
      | ^
In file included from /usr/include/gtk-2.0/gtk/gtktoolitem.h:31,
                 from /usr/include/gtk-2.0/gtk/gtktoolbutton.h:30,
                 from /usr/include/gtk-2.0/gtk/gtkmenutoolbutton.h:30,
                 from /usr/include/gtk-2.0/gtk/gtk.h:126,
                 from src/manipulation-gui/gui-watermark.c:3:
/usr/include/gtk-2.0/gtk/gtktooltips.h:73:3: warning: 'GTimeVal' is deprecated: Use 'GDateTime' instead [-Wdeprecated-declarations]
   73 |   GTimeVal last_popdown;
      |   ^~~~~~~~
In file included from /usr/include/glib-2.0/glib/galloca.h:32,
                 from /usr/include/glib-2.0/glib.h:30,
                 from /usr/include/glib-2.0/gobject/gbinding.h:28,
                 from /usr/include/glib-2.0/glib-object.h:22,
                 from /usr/include/glib-2.0/gio/gioenums.h:28,
                 from /usr/include/glib-2.0/gio/giotypes.h:28,
                 from /usr/include/glib-2.0/gio/gio.h:26,
                 from /usr/include/gtk-2.0/gdk/gdkapplaunchcontext.h:30,
                 from /usr/include/gtk-2.0/gdk/gdk.h:32,
                 from /usr/include/gtk-2.0/gtk/gtk.h:32,
                 from src/manipulation-gui/gui-watermark.c:3:
/usr/include/glib-2.0/glib/gtypes.h:547:8: note: declared here
  547 | struct _GTimeVal
      |        ^~~~~~~~~
/usr/bin/ld: /tmp/ccM6UWJC.o:(.bss+0x0): multiple definition of `bimp_window_main'; /tmp/ccNQVNQB.o:(.bss+0xa0): first defined here
/usr/bin/ld: /tmp/ccM6UWJC.o:(.bss+0x8): multiple definition of `bimp_selected_manipulations'; /tmp/ccNQVNQB.o:(.bss+0xa8): first defined here
/usr/bin/ld: /tmp/ccKKCFOC.o:(.bss+0x0): multiple definition of `bimp_selected_manipulations'; /tmp/ccNQVNQB.o:(.bss+0xa8): first defined here
/usr/bin/ld: /tmp/ccq22vBD.o:(.bss+0xb8): multiple definition of `bimp_input_filenames'; /tmp/ccNQVNQB.o:(.bss+0xd8): first defined here
/usr/bin/ld: /tmp/ccq22vBD.o:(.bss+0xa4): multiple definition of `bimp_opt_keepfolderhierarchy'; /tmp/ccNQVNQB.o:(.bss+0xc4): first defined here
/usr/bin/ld: /tmp/ccq22vBD.o:(.bss+0x88): multiple definition of `bimp_selected_manipulations'; /tmp/ccNQVNQB.o:(.bss+0xa8): first defined here
/usr/bin/ld: /tmp/ccq22vBD.o:(.bss+0xb0): multiple definition of `bimp_output_folder'; /tmp/ccNQVNQB.o:(.bss+0xd0): first defined here
/usr/bin/ld: /tmp/ccq22vBD.o:(.bss+0xa8): multiple definition of `bimp_opt_alertoverwrite'; /tmp/ccNQVNQB.o:(.bss+0xc8): first defined here
/usr/bin/ld: /tmp/ccq22vBD.o:(.bss+0x9c): multiple definition of `bimp_opt_keepdates'; /tmp/ccNQVNQB.o:(.bss+0xbc): first defined here
/usr/bin/ld: /tmp/ccq22vBD.o:(.bss+0x98): multiple definition of `bimp_is_busy'; /tmp/ccNQVNQB.o:(.bss+0xb8): first defined here
/usr/bin/ld: /tmp/ccq22vBD.o:(.bss+0x80): multiple definition of `bimp_window_main'; /tmp/ccNQVNQB.o:(.bss+0xa0): first defined here
/usr/bin/ld: /tmp/ccq22vBD.o:(.bss+0x90): multiple definition of `bimp_supported_procedures'; /tmp/ccNQVNQB.o:(.bss+0xb0): first defined here
/usr/bin/ld: /tmp/ccq22vBD.o:(.bss+0xa0): multiple definition of `bimp_opt_deleteondone'; /tmp/ccNQVNQB.o:(.bss+0xc0): first defined here
/usr/bin/ld: /tmp/cc0GTU6D.o:(.bss+0x8): multiple definition of `bimp_selected_manipulations'; /tmp/ccNQVNQB.o:(.bss+0xa8): first defined here
/usr/bin/ld: /tmp/cc0GTU6D.o:(.bss+0x10): multiple definition of `bimp_supported_procedures'; /tmp/ccNQVNQB.o:(.bss+0xb0): first defined here
/usr/bin/ld: /tmp/cc0GTU6D.o:(.bss+0x18): multiple definition of `bimp_is_busy'; /tmp/ccNQVNQB.o:(.bss+0xb8): first defined here
/usr/bin/ld: /tmp/cc0GTU6D.o:(.bss+0x1c): multiple definition of `bimp_opt_keepdates'; /tmp/ccNQVNQB.o:(.bss+0xbc): first defined here
/usr/bin/ld: /tmp/cc0GTU6D.o:(.bss+0x20): multiple definition of `bimp_opt_deleteondone'; /tmp/ccNQVNQB.o:(.bss+0xc0): first defined here
/usr/bin/ld: /tmp/cc0GTU6D.o:(.bss+0x24): multiple definition of `bimp_opt_keepfolderhierarchy'; /tmp/ccNQVNQB.o:(.bss+0xc4): first defined here
/usr/bin/ld: /tmp/cc0GTU6D.o:(.bss+0x28): multiple definition of `bimp_opt_alertoverwrite'; /tmp/ccNQVNQB.o:(.bss+0xc8): first defined here
/usr/bin/ld: /tmp/cc0GTU6D.o:(.bss+0x30): multiple definition of `bimp_output_folder'; /tmp/ccNQVNQB.o:(.bss+0xd0): first defined here
/usr/bin/ld: /tmp/cc0GTU6D.o:(.bss+0x38): multiple definition of `bimp_input_filenames'; /tmp/ccNQVNQB.o:(.bss+0xd8): first defined here
/usr/bin/ld: /tmp/ccWcRAPF.o:(.bss+0x0): multiple definition of `bimp_selected_manipulations'; /tmp/ccNQVNQB.o:(.bss+0xa8): first defined here
/usr/bin/ld: /tmp/ccun4SRE.o:(.bss+0x10): multiple definition of `bimp_supported_procedures'; /tmp/ccNQVNQB.o:(.bss+0xb0): first defined here
/usr/bin/ld: /tmp/ccun4SRE.o:(.bss+0x0): multiple definition of `bimp_window_main'; /tmp/ccNQVNQB.o:(.bss+0xa0): first defined here
/usr/bin/ld: /tmp/ccun4SRE.o:(.bss+0x8): multiple definition of `bimp_selected_manipulations'; /tmp/ccNQVNQB.o:(.bss+0xa8): first defined here
/usr/bin/ld: /tmp/ccun4SRE.o:(.bss+0x18): multiple definition of `bimp_is_busy'; /tmp/ccNQVNQB.o:(.bss+0xb8): first defined here
/usr/bin/ld: /tmp/ccun4SRE.o:(.bss+0x1c): multiple definition of `bimp_opt_keepdates'; /tmp/ccNQVNQB.o:(.bss+0xbc): first defined here
/usr/bin/ld: /tmp/ccun4SRE.o:(.bss+0x20): multiple definition of `bimp_opt_deleteondone'; /tmp/ccNQVNQB.o:(.bss+0xc0): first defined here
/usr/bin/ld: /tmp/ccun4SRE.o:(.bss+0x24): multiple definition of `bimp_opt_keepfolderhierarchy'; /tmp/ccNQVNQB.o:(.bss+0xc4): first defined here
/usr/bin/ld: /tmp/ccun4SRE.o:(.bss+0x28): multiple definition of `bimp_opt_alertoverwrite'; /tmp/ccNQVNQB.o:(.bss+0xc8): first defined here
/usr/bin/ld: /tmp/ccun4SRE.o:(.bss+0x30): multiple definition of `bimp_output_folder'; /tmp/ccNQVNQB.o:(.bss+0xd0): first defined here
/usr/bin/ld: /tmp/ccun4SRE.o:(.bss+0x38): multiple definition of `bimp_input_filenames'; /tmp/ccNQVNQB.o:(.bss+0xd8): first defined here
/usr/bin/ld: /tmp/ccs3XnoB.o:(.bss+0x1c8): multiple definition of `bimp_selected_manipulations'; /tmp/ccNQVNQB.o:(.bss+0xa8): first defined here
/usr/bin/ld: /tmp/ccgkRuSC.o:(.bss+0x30): multiple definition of `bimp_selected_manipulations'; /tmp/ccNQVNQB.o:(.bss+0xa8): first defined here
/usr/bin/ld: /tmp/cciVt7xB.o:(.bss+0x60): multiple definition of `bimp_selected_manipulations'; /tmp/ccNQVNQB.o:(.bss+0xa8): first defined here
/usr/bin/ld: /tmp/cct6WMLE.o:(.bss+0x18): multiple definition of `bimp_selected_manipulations'; /tmp/ccNQVNQB.o:(.bss+0xa8): first defined here
/usr/bin/ld: /tmp/ccmP66FF.o:(.bss+0x10): multiple definition of `bimp_window_main'; /tmp/ccNQVNQB.o:(.bss+0xa0): first defined here
/usr/bin/ld: /tmp/ccmP66FF.o:(.bss+0x18): multiple definition of `bimp_selected_manipulations'; /tmp/ccNQVNQB.o:(.bss+0xa8): first defined here
/usr/bin/ld: /tmp/cctieY8B.o:(.bss+0x38): multiple definition of `spin_width'; /tmp/cciVt7xB.o:(.bss+0x18): first defined here
/usr/bin/ld: /tmp/cctieY8B.o:(.bss+0x30): multiple definition of `spin_height'; /tmp/cciVt7xB.o:(.bss+0x10): first defined here
/usr/bin/ld: /tmp/cctieY8B.o:(.bss+0x78): multiple definition of `bimp_selected_manipulations'; /tmp/ccNQVNQB.o:(.bss+0xa8): first defined here
/usr/bin/ld: /tmp/cc1GpBrF.o:(.bss+0x8): multiple definition of `bimp_selected_manipulations'; /tmp/ccNQVNQB.o:(.bss+0xa8): first defined here
/usr/bin/ld: /tmp/cc3NRYwC.o:(.bss+0x40): multiple definition of `bimp_supported_procedures'; /tmp/ccNQVNQB.o:(.bss+0xb0): first defined here
/usr/bin/ld: /tmp/cc3NRYwC.o:(.bss+0x38): multiple definition of `bimp_window_main'; /tmp/ccNQVNQB.o:(.bss+0xa0): first defined here
/usr/bin/ld: /tmp/cc3NRYwC.o:(.bss+0x48): multiple definition of `bimp_is_busy'; /tmp/ccNQVNQB.o:(.bss+0xb8): first defined here
/usr/bin/ld: /tmp/cc3NRYwC.o:(.bss+0x4c): multiple definition of `bimp_opt_keepdates'; /tmp/ccNQVNQB.o:(.bss+0xbc): first defined here
/usr/bin/ld: /tmp/cc3NRYwC.o:(.bss+0x50): multiple definition of `bimp_opt_deleteondone'; /tmp/ccNQVNQB.o:(.bss+0xc0): first defined here
/usr/bin/ld: /tmp/cc3NRYwC.o:(.bss+0x54): multiple definition of `bimp_opt_keepfolderhierarchy'; /tmp/ccNQVNQB.o:(.bss+0xc4): first defined here
/usr/bin/ld: /tmp/cc3NRYwC.o:(.bss+0x58): multiple definition of `bimp_opt_alertoverwrite'; /tmp/ccNQVNQB.o:(.bss+0xc8): first defined here
/usr/bin/ld: /tmp/cc3NRYwC.o:(.bss+0x60): multiple definition of `bimp_output_folder'; /tmp/ccNQVNQB.o:(.bss+0xd0): first defined here
/usr/bin/ld: /tmp/cc3NRYwC.o:(.bss+0x68): multiple definition of `bimp_input_filenames'; /tmp/ccNQVNQB.o:(.bss+0xd8): first defined here
/usr/bin/ld: /tmp/cc3NRYwC.o:(.bss+0x70): multiple definition of `bimp_selected_manipulations'; /tmp/ccNQVNQB.o:(.bss+0xa8): first defined here
/usr/bin/ld: /tmp/ccaYMnhC.o:(.bss+0xb8): multiple definition of `bimp_selected_manipulations'; /tmp/ccNQVNQB.o:(.bss+0xa8): first defined here
collect2: error: ld returned 1 exit status
make: *** [Makefile:6: make] Error 1
```
In my case the compiler is GCC 10.1.0. Dunno if it also happened with the 9.x one.

--- a/src/bimp-gui.c
+++ b/src/bimp-gui.c
@@ -55,18 +55,20 @@ static void progressbar_end_hidden (gpointer);
 static void progressbar_settext_hidden (const gchar*, gpointer);
 static void progressbar_setvalue_hidden (double, gpointer);
 
-GtkWidget *panel_sequence, *panel_options;
-GtkWidget *hbox_sequence;
-GtkWidget *scroll_sequence;
-GtkWidget *popmenu_add, *popmenu_edit, *popmenu_addfiles, *popmenu_removefiles;
-GtkWidget *check_keepfolderhierarchy, *check_deleteondone, *check_keepdates;
-GtkWidget *treeview_files;
-GtkWidget *button_preview, *button_outfolder, *button_samefolder;
-GtkWidget* progressbar_visible;
+GtkWidget* bimp_window_main;
 
-char* selected_source_folder;
-char* last_input_location;
-const gchar* progressbar_data;
+static GtkWidget *panel_sequence, *panel_options;
+static GtkWidget *hbox_sequence;
+static GtkWidget *scroll_sequence;
+static GtkWidget *popmenu_add, *popmenu_edit, *popmenu_addfiles, *popmenu_removefiles;
+static GtkWidget *check_keepfolderhierarchy, *check_deleteondone, *check_keepdates;
+static GtkWidget *treeview_files;
+static GtkWidget *button_preview, *button_outfolder, *button_samefolder;
+static GtkWidget* progressbar_visible;
+
+static char* selected_source_folder;
+static char* last_input_location;
+static const gchar* progressbar_data;
 
 enum /* TreeView stuff... */
 {
@@ -74,7 +76,7 @@ enum /* TreeView stuff... */
   N_COLUMNS
 };
 
-manipulation clicked_man; /* temporary manipulation, selected by clicking on panel_seq buttons */
+static manipulation clicked_man; /* temporary manipulation, selected by clicking on panel_seq buttons */
 
 void bimp_show_gui() 
 {	

--- a/src/bimp-gui.h
+++ b/src/bimp-gui.h
@@ -21,6 +21,6 @@ void bimp_progress_bar_set(double, char*);
 void bimp_show_error_dialog(char*, GtkWidget*);
 void bimp_set_busy(gboolean);
 
-GtkWidget* bimp_window_main;
+extern GtkWidget* bimp_window_main;
 
 #endif

--- a/src/bimp-manipulations.c
+++ b/src/bimp-manipulations.c
@@ -8,6 +8,9 @@
 #include "bimp-manipulations-gui.h"
 #include "plugin-intl.h"
 
+
+GSList* bimp_selected_manipulations;
+
 /* Appends a default manipulation to the step list */
 manipulation bimp_append_manipulation(manipulation_type type)
 {

--- a/src/bimp-manipulations.h
+++ b/src/bimp-manipulations.h
@@ -278,6 +278,6 @@ manipulation manipulation_changeformat_new(void);
 manipulation manipulation_rename_new(void); 
 manipulation manipulation_userdef_new(void); 
 
-GSList* bimp_selected_manipulations; /* Manipulations selected by user */
+extern GSList* bimp_selected_manipulations; /* Manipulations selected by user */
 
 #endif

--- a/src/bimp-operate.c
+++ b/src/bimp-operate.c
@@ -43,31 +43,31 @@ static gboolean image_save_exr(image_output);
 
 static int overwrite_result(char*, GtkWidget*);
 
-char* current_datetime;
-int processed_count;
-int success_count;
-int total_images;
+static char* current_datetime;
+static int processed_count;
+static int success_count;
+static int total_images;
 
-char* common_folder_path;
+static char* common_folder_path;
 
-gboolean list_contains_changeformat;
-gboolean list_contains_rename;
-gboolean list_contains_watermark;
-gboolean list_contains_savingplugin;
+static gboolean list_contains_changeformat;
+static gboolean list_contains_rename;
+static gboolean list_contains_watermark;
+static gboolean list_contains_savingplugin;
 
 // set of variables to be used when doing Curve color correction
 // they are global so the batch process will read the source curve file once
-gboolean colorcurve_init;
-int colorcurve_num_points_v;
-gdouble* colorcurve_ctr_points_v;
-int colorcurve_num_points_r;
-gdouble* colorcurve_ctr_points_r;
-int colorcurve_num_points_g;
-gdouble* colorcurve_ctr_points_g;
-int colorcurve_num_points_b;
-gdouble* colorcurve_ctr_points_b;
-int colorcurve_num_points_a;
-gdouble* colorcurve_ctr_points_a;
+static gboolean colorcurve_init;
+static int colorcurve_num_points_v;
+static gdouble* colorcurve_ctr_points_v;
+static int colorcurve_num_points_r;
+static gdouble* colorcurve_ctr_points_r;
+static int colorcurve_num_points_g;
+static gdouble* colorcurve_ctr_points_g;
+static int colorcurve_num_points_b;
+static gdouble* colorcurve_ctr_points_b;
+static int colorcurve_num_points_a;
+static gdouble* colorcurve_ctr_points_a;
 
 void bimp_start_batch(gpointer parent_dialog)
 {

--- a/src/bimp-serialize.c
+++ b/src/bimp-serialize.c
@@ -36,8 +36,8 @@ static manipulation read_rename(GKeyFile*);
 static void write_userdef(userdef_settings, GKeyFile*, int);
 static manipulation read_userdef(GKeyFile*, int);
 
-int userdef_count;
-int loaded_build;
+static int userdef_count;
+static int loaded_build;
 
 gboolean bimp_serialize_to_file(gchar* filename)
 {

--- a/src/bimp.c
+++ b/src/bimp.c
@@ -32,6 +32,18 @@
 #include "bimp-utils.h"
 #include "plugin-intl.h"
 
+GSList* bimp_input_filenames;
+char* bimp_output_folder;
+
+gint bimp_opt_alertoverwrite;
+gboolean bimp_opt_keepfolderhierarchy;
+gboolean bimp_opt_deleteondone;
+gboolean bimp_opt_keepdates;
+
+gboolean bimp_is_busy;
+
+GSList* bimp_supported_procedures;
+
 static void query (void);
 static gboolean pdb_proc_has_compatible_params (gchar*);
 
@@ -43,7 +55,7 @@ static void run (
     GimpParam **return_vals
     );
 
-const GimpPlugInInfo PLUG_IN_INFO = {
+static const GimpPlugInInfo PLUG_IN_INFO = {
     NULL,  /* init_proc  */
     NULL,  /* quit_proc  */
     query, /* query_proc */

--- a/src/bimp.h
+++ b/src/bimp.h
@@ -20,17 +20,17 @@
 #define BIMP_OVERWRITE_SKIP_ASK 1
 #define BIMP_DONT_OVERWRITE_SKIP_ASK 2
 
-GSList* bimp_input_filenames;
-char* bimp_output_folder;
+extern GSList* bimp_input_filenames;
+extern char* bimp_output_folder;
 
-gint bimp_opt_alertoverwrite;
-gboolean bimp_opt_keepfolderhierarchy;
-gboolean bimp_opt_deleteondone;
-gboolean bimp_opt_keepdates;
+extern gint bimp_opt_alertoverwrite;
+extern gboolean bimp_opt_keepfolderhierarchy;
+extern gboolean bimp_opt_deleteondone;
+extern gboolean bimp_opt_keepdates;
 
-gboolean bimp_is_busy;
+extern gboolean bimp_is_busy;
 
-GSList* bimp_supported_procedures;
+extern GSList* bimp_supported_procedures;
 void init_supported_procedures(void);
 
 #endif

--- a/src/manipulation-gui/gui-changeformat.c
+++ b/src/manipulation-gui/gui-changeformat.c
@@ -8,17 +8,17 @@ static void update_frame_params(GtkComboBox*, changeformat_settings);
 static void adv_expanded (GtkWidget*, GtkRequisition*, gpointer);
 static void update_window_size();
 
-GtkWidget *frame_params, *inner_widget;
-GtkWidget *combo_format, *scale_quality, *scale_alpha_quality, *scale_smoothing, *check_interlace, *scale_compression, *check_baseline;
-GtkWidget *check_rle, *check_progressive, *check_entrophy, *combo_compression, *spin_markers, *combo_subsampling, *combo_dct, *combo_origin;
-GtkWidget *check_savebgc, *check_savegamma, *check_saveoff, *check_savephys, *check_savetime, *check_savecomm, *check_savetrans, *check_lossless;
-GtkWidget *combo_preset, *check_saveexif, *check_savexmp, *check_savecp;
-GtkWidget *expander_advanced;
+static GtkWidget *frame_params, *inner_widget;
+static GtkWidget *combo_format, *scale_quality, *scale_alpha_quality, *scale_smoothing, *check_interlace, *scale_compression, *check_baseline;
+static GtkWidget *check_rle, *check_progressive, *check_entrophy, *combo_compression, *spin_markers, *combo_subsampling, *combo_dct, *combo_origin;
+static GtkWidget *check_savebgc, *check_savegamma, *check_saveoff, *check_savephys, *check_savetime, *check_savecomm, *check_savetrans, *check_lossless;
+static GtkWidget *combo_preset, *check_saveexif, *check_savexmp, *check_savecp;
+static GtkWidget *expander_advanced;
 
-GtkTextBuffer *buffer_comment;
-GtkTextIter start_comment, end_comment;
+static GtkTextBuffer *buffer_comment;
+static GtkTextIter start_comment, end_comment;
 
-GtkWidget* parentwin;
+static GtkWidget* parentwin;
 
 GtkWidget* bimp_changeformat_gui_new(changeformat_settings settings, GtkWidget* parent)
 {

--- a/src/manipulation-gui/gui-color.c
+++ b/src/manipulation-gui/gui-color.c
@@ -6,9 +6,9 @@
 
 static void toggle_curve(GtkToggleButton*, gpointer);
 
-GtkWidget *scale_bright, *scale_contrast;
-GtkWidget *check_autolevels, *check_grayscale, *check_curve;
-GtkWidget *chooser_curve;
+static GtkWidget *scale_bright, *scale_contrast;
+static GtkWidget *check_autolevels, *check_grayscale, *check_curve;
+static GtkWidget *chooser_curve;
     
 GtkWidget* bimp_color_gui_new(color_settings settings)
 {

--- a/src/manipulation-gui/gui-crop.c
+++ b/src/manipulation-gui/gui-crop.c
@@ -8,11 +8,11 @@ static void toggle_group(GtkToggleButton*, gpointer);
 static void set_customratio(GtkComboBox*, gpointer);
 static char* crop_preset_get_string(crop_preset);
 
-GtkWidget *hbox_ratio, *table_manual, *hbox_customratio;
-GtkWidget *radio_stratio, *radio_manual;
-GtkWidget *combo_ratio, *spin_ratio1, *spin_ratio2;
-GtkWidget *spin_width, *spin_height;
-GtkWidget *hbox_startpos, *combo_startpos;
+static GtkWidget *hbox_ratio, *table_manual, *hbox_customratio;
+static GtkWidget *radio_stratio, *radio_manual;
+static GtkWidget *combo_ratio, *spin_ratio1, *spin_ratio2;
+static GtkWidget *spin_width, *spin_height;
+static GtkWidget *hbox_startpos, *combo_startpos;
 
 GtkWidget* bimp_crop_gui_new(crop_settings settings)
 {

--- a/src/manipulation-gui/gui-fliprotate.c
+++ b/src/manipulation-gui/gui-fliprotate.c
@@ -7,7 +7,7 @@
 #include "../bimp-utils.h"
 #include "../plugin-intl.h"
     
-GtkWidget *button_flipH, *button_flipV, *combo_rotate;
+static GtkWidget *button_flipH, *button_flipV, *combo_rotate;
 
 GtkWidget* bimp_fliprotate_gui_new(fliprotate_settings settings)
 {

--- a/src/manipulation-gui/gui-rename.c
+++ b/src/manipulation-gui/gui-rename.c
@@ -9,7 +9,7 @@
 #include "../plugin-intl.h"
 
 static void check_entrytext (GtkEditable*, gpointer);
-GtkWidget *entry_pattern, *label_preview;
+static GtkWidget *entry_pattern, *label_preview;
 
 GtkWidget* bimp_rename_gui_new(rename_settings settings, GtkWidget* parent)
 {

--- a/src/manipulation-gui/gui-resize.c
+++ b/src/manipulation-gui/gui-resize.c
@@ -11,17 +11,17 @@ static void toggle_resolution(GtkToggleButton*, gpointer);
 static void update_dimensions();
 
 
-GtkWidget *table_res;
-GtkWidget *check_resolution;
-GtkWidget *combo_unitW, *combo_unitH, *combo_aspectratio, *combo_quality;
-GtkWidget *chooser_paddingcolor;
-GtkWidget *spin_width, *spin_height, *spin_resX, *spin_resY;
+static GtkWidget *table_res;
+static GtkWidget *check_resolution;
+static GtkWidget *combo_unitW, *combo_unitH, *combo_aspectratio, *combo_quality;
+static GtkWidget *chooser_paddingcolor;
+static GtkWidget *spin_width, *spin_height, *spin_resX, *spin_resY;
 
-resize_mode previous_resize_w, previous_resize_h;
-gdouble last_percent_w_value;
-gdouble last_percent_h_value;
-gint last_pixel_w_value;
-gint last_pixel_h_value;
+static resize_mode previous_resize_w, previous_resize_h;
+static gdouble last_percent_w_value;
+static gdouble last_percent_h_value;
+static gint last_pixel_w_value;
+static gint last_pixel_h_value;
 
 GtkWidget* bimp_resize_gui_new(resize_settings settings)
 {

--- a/src/manipulation-gui/gui-sharpblur.c
+++ b/src/manipulation-gui/gui-sharpblur.c
@@ -4,7 +4,7 @@
 #include "../bimp-manipulations.h"
 #include "../plugin-intl.h"
 
-GtkWidget *scale_sharpblur;
+static GtkWidget *scale_sharpblur;
 
 GtkWidget* bimp_sharpblur_gui_new(sharpblur_settings settings)
 {

--- a/src/manipulation-gui/gui-userdef.c
+++ b/src/manipulation-gui/gui-userdef.c
@@ -16,14 +16,14 @@ static gboolean select_procedure (GtkTreeSelection*, GtkTreeModel*, GtkTreePath*
 static void update_selected_procedure(gchar*);
 static void update_procedure_box(userdef_settings);
 
-GtkWidget *treeview_procedures;
-GtkWidget *scroll_procparam, *vbox_procparam;
-GtkWidget *parent_dialog;
+static GtkWidget *treeview_procedures;
+static GtkWidget *scroll_procparam, *vbox_procparam;
+static GtkWidget *parent_dialog;
 
-userdef_settings temp_settings;
-GtkWidget **param_widget; /* array of showable widgets for customizing the selected procedure */
+static userdef_settings temp_settings;
+static GtkWidget **param_widget; /* array of showable widgets for customizing the selected procedure */
 
-GtkTreeSelection *treesel_proc;
+static GtkTreeSelection *treesel_proc;
 
 enum
 {

--- a/src/manipulation-gui/gui-watermark.c
+++ b/src/manipulation-gui/gui-watermark.c
@@ -13,13 +13,13 @@ static const char* watermark_pos_get_string(watermark_position);
 static const char* watermark_pos_get_abbreviation(watermark_position);
 static void file_filters_add_patterns(GtkFileFilter*, GtkFileFilter*, ...);
 
-GtkWidget *table_text, *vbox_image, *hbox_image_size;
-GtkWidget *radio_text, *radio_image;
-GtkWidget *entry_text;
-GtkWidget *chooser_font, *chooser_color, *chooser_image;
-GtkWidget *check_image_adaptsize, *spin_image_sizepercent, *spin_edge, *combo_image_sizemode;
-GtkWidget *scale_opacity;
-GtkWidget *position_buttons[9];
+static GtkWidget *table_text, *vbox_image, *hbox_image_size;
+static GtkWidget *radio_text, *radio_image;
+static GtkWidget *entry_text;
+static GtkWidget *chooser_font, *chooser_color, *chooser_image;
+static GtkWidget *check_image_adaptsize, *spin_image_sizepercent, *spin_edge, *combo_image_sizemode;
+static GtkWidget *scale_opacity;
+static GtkWidget *position_buttons[9];
 
 GtkWidget* bimp_watermark_gui_new(watermark_settings settings)
 {

